src/policy.test.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { TESTNET, TEST_FIXTURES, BraidDetails } from "unchained-bitcoin";
+import {
+  KeyOrigin,
+  validateMultisigPolicyTemplate,
+  getPolicyTemplateFromWalletConfig,
+  braidDetailsToWalletConfig,
+} from "./policy";
+import { POLICY_FIXTURE } from "./fixtures";
+
+describe("validateMultisigPolicyTemplate", () => {
+  it("throws error if script type is not supported", () => {
+    const templates = [
+      "pkh(@0/**)",
+      "tr(@0/**)",
+      "wpkh(@0/**)",
+      "foobar(@0/**)",
+      "(@0/**)",
+    ];
+
+    for (const template of templates) {
+      expect(() => validateMultisigPolicyTemplate(template)).toThrowError();
+    }
+  });
+
+  it("throws if required signers is invalid", () => {
+    const templates = [
+      "wsh(3,@0/**,@1/**)", // not enough signers
+      "sh(@0/**)", // no required signers indicated
+      "wsh()", // no keys or required signers
+      "sh(2,@0,@1/**)", // policy is missing derivation wildcard
+    ];
+
+    for (const template of templates) {
+      expect(() => validateMultisigPolicyTemplate(template)).toThrowError();
+    }
+  });
+
+  it("does not throw if templates are valid", () => {
+    const templates = [
+      "wsh(2,@0/**,@1/**)",
+      "sh(2,@0/**,@1/**)",
+      "sh(2,@0/**,@1/**,@2/**)",
```

Comment:
out of curiosity is `sh` just a abbreviated version of `p2sh` ?

Reply:
yep! That's the descriptor/policy language. Basically I think the idea is that `pay-to-` part isn't really relevant to the actual descriptor we're talking about.